### ISSUE TITLE
Rule Change: `no-empty` detect non-existant blocks as well

### ISSUE THREAD

**Issue by silverwind** [eslint/eslint#17357]
### What rule do you want to change?

[`no-empty`](https://eslint.org/docs/latest/rules/no-empty)

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
if (false);
while (false);
```

[Playground Example](https://eslint.org/play/#eyJ0ZXh0IjoiaWYgKGZhbHNlKTtcbndoaWxlIChmYWxzZSk7XG5pZiAoZmFsc2UpIHt9XG53aGlsZSAoZmFsc2UpIHt9Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJuby1lbXB0eSI6WyJlcnJvciJdfSwiZW52Ijp7fX19)

### What does the rule currently do for this code?

It ignores the empty `if` and `while` bodies.

### What will the rule do after it's changed?

It will report the empty `if`and `while` bodies.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

The rule description is "Disallow empty block statements", and I think stricly speaking, `if (condition);` is not a block statement but I think the intent of the rule would fit quite well to extend it.

**Comment by sam3k** [comment]
Hi @silverwind, thank you for providing a playground demo. I think this is a valid edge case and will move it forward.

**Comment by nzakas** [comment]
The intent of this rule is make it easy to see when you have empty blocks, not when loops have empty statements in general. In particular, it's easy to start with a block that has code in it and then accidentally end up with an empty block by deleting lines. Non-existent blocks, as in your examples, are unlikely to be accidental.

So, I don't think this fit with the `no-empty` rule and I'd suggest creating your own rule for this particular case.

**Comment by silverwind** [comment]
> Non-existent blocks, as in your examples, are unlikely to be accidental.

I had this exact bug where the intention was `if (cond) return;` but I ended up forgetting the `return`, making the statement a noop. Would you consider such a rule a candidate for eslint core rules?

**Comment by sam3k** [comment]
> > Non-existent blocks, as in your examples, are unlikely to be accidental.
> 
> I had this exact bug where the intention was `if (cond) return;` but I forgot the `return`, making the statement a noop.
> 
> Would you consider such a rule a candidate for a core rule `no-missing-statement`?

I am not aware of a `no-missing-statement` rule. At this point, if you have any other issues, I recommend opening a new issue since this one is closed and not relevant.

**Comment by silverwind** [comment]
> I am not aware of a no-missing-statement rule

I'm proposing a new rule, in case that is not clear.

**Comment by sam3k** [comment]
@silverwind understood. Thanks for clarifying. If you would like to propose a new rule, you will have to create a new issue for that.
